File: Quick-Sort.py
def QuickSort(array):
    if len(array) <= 1:
        return array
    else:
        pi_index = partition(array)
        return QuickSort(array[:pi_index]) + [array[pi_index]] + QuickSort(array[pi_index + 1:])

def partition(array):
    pivot = array[0]
    i = j = 1
    while j < len(array):
        if array[j] > pivot:
            j += 1
        else:
            array[i], array[j] = array[j], array[i]
            i += 1
            j += 1
    array[0], array[i - 1] = array[i - 1], array[0]
    return i - 1

File: Quick-Sort.c
#include <stdio.h>

int partition(int *arr, int n, int low, int high){
    int pivot = arr[low];
    int temp;
    int i = 1;
    int j = 1;
    while(j <= high){
        if(arr[j] < pivot){
            temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            i++;
            j++;
        }
        else{
            j++;
        }
    }
    i--;
    temp = arr[0];
    arr[0] = arr[i];
    arr[i] = temp;
    return i;
}

void Quicksort(int *arr, int n, int low, int high){
    if(n <= 1){
        return;
    }
    int pi_index= partition(arr, n, low, high);
    Quicksort(arr, n, 0, pi_index);
    Quicksort(arr, n, pi_index, n);
}